server/aesdsocket.c

```diff
@@ -303,11 +319,13 @@ int main(int argc, char *argv[])
         inet_ntop( AF_INET, &ipAddr,str, INET_ADDRSTRLEN);
         syslog(LOG_DEBUG, "Accepted connection to %s\n", str);
 
-        
-        if(!first_time)
+        if(!(ALARM_AESD_CHAR))
         {
-            alarm(10);
-            first_time = 1;
+            if(!first_time)
+            {
+                alarm(10);
+                first_time = 1;
+            }
```

Comment:
@Jahnavi-pinnamaneni what is this logic doing here?

Reply:
@swatikadivar That was redundant code I think, I used that when we had to timestamp, so that, the alarm gets its first trigger and then further on calls itself recurcively